app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt

```diff
@@ -0,0 +1,327 @@
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import ch.hikemate.app.R
+import ch.hikemate.app.model.route.HikeRoute
+import ch.hikemate.app.model.route.ListOfHikeRoutesViewModel
+import ch.hikemate.app.ui.components.AppropriatenessMessage
+import ch.hikemate.app.ui.components.BackButton
+import ch.hikemate.app.ui.navigation.NavigationActions
+import ch.hikemate.app.ui.navigation.Screen
+import ch.hikemate.app.utils.calculateBestZoomLevel
+import ch.hikemate.app.utils.getGeographicalCenter
+import com.google.firebase.Timestamp
+import java.util.Calendar
+import org.osmdroid.config.Configuration
+import org.osmdroid.views.CustomZoomButtonsController
+import org.osmdroid.views.MapView
+
+const val MAP_MIN_ZOOM = 3.0
+const val MAP_MAX_ZOOM = 18.0
+
+const val TEST_TAG_MAP = "map"
+const val TEST_TAG_HIKE_NAME = "hikeName"
+const val TEST_TAG_BOOKMARK_ICON = "bookmarkIcon"
+const val TEST_TAG_ELEVATION_GRAPH = "elevationGraph"
+const val TEST_TAG_DETAIL_ROW_TAG = "detailRowTag"
+const val TEST_TAG_DETAIL_ROW_VALUE = "detailRowValue"
+const val TEST_TAG_ADD_DATE_BUTTON = "addDateButton"
+const val TEST_TAG_PLANNED_DATE_TEXT_BOX = "plannedDateTextBox"
+
+@Composable
+fun HikeDetailScreen(
+    listOfHikeRoutesViewModel: ListOfHikeRoutesViewModel,
+    navigationActions: NavigationActions
+) {
+
+  val route: HikeRoute =
+      listOfHikeRoutesViewModel.selectedHikeRoute.collectAsState().value
+          ?: throw IllegalStateException(
+              "Selected HikeRoute is null. This screen should only be shown when a route is selected.")
+
+  val context = LocalContext.current
+
+  // Only do the configuration on the first composition, not on every recomposition
+  LaunchedEffect(Unit) {
+    Configuration.getInstance().apply {
+      // Set user-agent to avoid rejected requests
+      userAgentValue = context.packageName
+
+      // Allow for faster loading of tiles. Default OSMDroid value is 2.
+      tileDownloadThreads = 4
+
+      // Maximum number of tiles that can be downloaded at once. Default is 40.
+      tileDownloadMaxQueueSize = 40
+
+      // Maximum number of bytes that can be used by the tile file system cache. Default is 600MB.
+      tileFileSystemCacheMaxBytes = 600L * 1024L * 1024L
+    }
+  }
+
+  // Avoid re-creating the MapView on every recomposition
+  val mapView = remember {
+    MapView(context).apply {
+      // Set map's initial state
+      controller.setZoom(calculateBestZoomLevel(route.bounds))
```

Comment:
Change: this is marked as deprecated and Sonar does not like it. Apparently the signature of `setZoom` that takes an `Int` is deprecated, but not the one that takes a `Double`, so you can fix it by doing

```suggestion
      controller.setZoom(calculateBestZoomLevel(route.bounds).toDouble())
```